.github/workflows/vuln-check-reusable.yaml

```
@@ -0,0 +1,151 @@
+        required: false
+        type: boolean
+        default: false
+      images:
+        description: "JSON string representing a list of pairs of names and image base names of Docker images. For example, if `'[[\"ScalarDL Ledger\", \"scalar-ledger\"], [\"ScalarDL Auditor\", \"scalar-auditor\"]'` is specified, the scanner will scan `ghcr.io/scalar-labs/scalar-ledger:<tag>` and `ghcr.io/scalar-labs/scalar-auditor:<tag>`."
+        required: true
+        type: string
+      version-command:
+        description: 'Shell command that prints the version string. It is used as the tag for Docker images to be scanned.'
+        required: false
+        type: string
+        default: "./gradlew :common:properties -q | grep version: | awk '{print $2}'"
+    secrets:
+      CR_PAT:
+        required: true
+      SLACK_SECURITY_WEBHOOK_URL:
+        required: true
+
+env:
+  TERM: dumb
+  GPR_USERNAME: ${{ github.repository_owner }}
+  GPR_PASSWORD: ${{ secrets.CR_PAT }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    env:
+      GH_TOKEN: ${{ secrets.CR_PAT }}
+
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+
+    steps:
+      - name: Checkout vuln-check scripts
+        uses: actions/checkout@v4
+        with:
+          repository: scalar-labs/actions
+          sparse-checkout: vuln-check-script
+          ref: main
+          path: vuln-check
+
+      - id: prepare-target-ref
+        name: Prepare target-ref
+        env:
+          SCRIPT_PATH: ./vuln-check/vuln-check-script
+        run: |
+          # Find the latest release name if `find-latest-release` is set to true. Use the value of `target-ref` as is otherwise.
+          if [[ ${{ inputs.find-latest-release }} = 'true' ]]; then
+            releases=$($SCRIPT_PATH/fetch_gh_releases "${{ github.event.repository.owner.login }}" "${{ github.event.repository.name }}")
+            echo -------------
+            echo "releases: $releases"
+            echo -------------
+            target_release=''
+            if [[ -n $releases ]]; then
+              target_release=$($SCRIPT_PATH/find_latest_release "${{ inputs.target-ref }}" $releases)
+            fi
+            if [[ -z $target_release ]]; then
+              echo "Can't find a target release"
+              exit 1
+            fi
+            echo $target_release
+            echo "target-ref=$target_release" >> $GITHUB_OUTPUT
+          else
+            echo "target-ref=${{ inputs.target-ref }}" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Checkout the target repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ steps.prepare-target-ref.outputs.target-ref }}
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v4
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+
+      - name: Docker build
+        uses: gradle/gradle-build-action@v3
+        with:
+          arguments: docker
```

The `gradle-build-action` has been [deprecated](https://github.com/gradle/actions/blob/main/docs/deprecation-upgrade-guide.md), could you replace it with something like that ?

```suggestion
      - name: Setup Gradle
        uses: gradle/actions/setup-gradle@v4

      - name: Docker build
        run: ./gradlew docker
```

---

Yes, I noticed the deprecation warnings while running the workflow, but I was wondering if I should include the fix in the same PR, so I just copied the current working code. I would like to fix it right away in the next PR.

---

Allright!